app/crud/categories.php

```diff
@@ -1,245 +1,277 @@
+     aria-hidden="true">
+    <div class="modal-dialog" role="document">
+        <div class="modal-content">
+            <div class="modal-header">
+                <h5 class="modal-title" id="exampleModalLabel">Add Data</h5>
+                <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                    <span aria-hidden="true">&times;</span>
+                </button>
             </div>
-        </div>
-    </div>
 
-    <!-- EDIT POP UP FORM (Bootstrap MODAL) -->
-    <div class="modal fade" id="editmodal" tabindex="-1" role="dialog" aria-labelledby="exampleModalLabel"
-        aria-hidden="true">
-        <div class="modal-dialog" role="document">
-            <div class="modal-content">
-                <div class="modal-header">
-                    <h5 class="modal-title" id="exampleModalLabel">Edit Data</h5>
-                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
-                        <span aria-hidden="true">&times;</span>
-                    </button>
-                </div>
+            <form action="categories_operation.php" method="POST">
+                <div class="modal-body">
+                    <?php
+                        require_once '../models/Competition.php';
 
-                <form action="categories_operation.php" method="POST">
-                    <div class="modal-body">
-                        <input type="hidden" name="update_id" id="update_id">
-                        <div class="form-group">
-                            <label>Competition_ID</label>
-                            <input type="number" min="1" max="2" name="competition_id" id="competition_id" class="form-control" placeholder="Select your Competition_ID 1(sports) or 2(litmusda)" autocomplete="off" required>
-                        </div>
-
-                        <div class="form-group">
-                            <label>Slug</label>
-                            <input type="text" name="slug" id="slug" class="form-control" placeholder="Enter your Slug">
-                        </div>
-
-                        <div class="form-group">
-                            <label>Title</label>
-                            <input type="text" name="title" id="title" class="form-control" placeholder="Enter your Title">
-                        </div>
+                        $page_url = $_SERVER['REQUEST_URI'];
+                        $entity_category = '';
+                        $competitions = Competition::all();
+                        foreach ($competitions as $competition) {
+                            $competition_url = strtolower(str_replace(' ', '-', $competition->getSlug()));
+                            if (strpos($page_url, $competition_url) !== false) {
+                                $entity_category = $competition->getTitle();
+                                break;
+                            }
+                        }
```

Comment:
The "Add Data" modal inside your `categories.php` always defaults to "Sports" competition because your code is searching for the slug "sports" in the URL. But it so happened that the URL will always contain the word "sports": "localhost/***sports***fest-litmusda/..."

I think you can just use the value of `$_GET['competition']`, if it is present in the request, for your `$entity_category`.


```suggestion
                        $entity_category = isset($_GET['competition']) ? strtolower(trim($_GET['competition'])) : '';
                        $competitions = Competition::all();
```